cmd/kubeadm/app/cmd/phases/controlplanepreparejoin.go

```diff
@@ -0,0 +1,148 @@
+		Run:   runCertsPhaseLocal(),
+	}
+	return phase
+}
+
+func newKubeconfigSubphase() workflow.Phase {
+	phase := workflow.Phase{
+		Name:  "kubeconfig",
+		Short: "Generates the kubeconfig for all new control plane components",
+		Run:   runKubeconfigPhaseLocal(),
+	}
+	return phase
+}
+
+func newManifestsSubphase() workflow.Phase {
+	phase := workflow.Phase{
+		Name:  "manifests",
+		Short: "Generates the manifests for all the new control plane components",
+		Run:   runManifestsPhaseLocal(),
+	}
+	return phase
+}
+
+func runCertsPhaseLocal() func(c workflow.RunData) error {
+	return func(c workflow.RunData) error {
+		data, ok := c.(controlPlanePrepareData)
+		if !ok {
+			return errors.New("certs phase invoked with an invalid data struct")
+		}
+		cfg := data.Cfg()
+
+		// Generate missing certificates (if any)
+		if err := certsphase.CreatePKIAssets(cfg); err != nil {
+			return err
+		}
+
+		return nil
+	}
+}
+
+func runKubeconfigPhaseLocal() func(c workflow.RunData) error {
+	return func(c workflow.RunData) error {
+		data, ok := c.(controlPlanePrepareData)
+		if !ok {
+			return errors.New("kubeconfig phase invoked with an invalid data struct")
+		}
+		cfg := data.Cfg()
+
+		// Generate kubeconfig files for controller manager, scheduler and for the admin/kubeadm itself
+		// NB. The kubeconfig file for kubelet will be generated by the TLS bootstrap process in
+		// following steps of the join --experimental-control plane workflow
+		if err := kubeconfigphase.CreateJoinControlPlaneKubeConfigFiles(kubeadmconstants.KubernetesDir, cfg); err != nil {
+			return errors.Wrap(err, "error generating kubeconfig files")
+		}
+
+		return nil
+	}
+}
+
+func runManifestsPhaseLocal() func(c workflow.RunData) error {
+	return func(c workflow.RunData) error {
+		data, ok := c.(controlPlanePrepareData)
+		if !ok {
+			return errors.New("manifests phase invoked with an invalid data struct")
+		}
+		cfg := data.Cfg()
+
+		// Creates static pod manifests file for the control plane components to be deployed on this node
+		// Static pods will be created and managed by the kubelet as soon as it starts
+		if err := controlplanephase.CreateInitStaticPodManifestFiles(kubeadmconstants.GetStaticPodDirectory(), cfg); err != nil {
+			return errors.Wrap(err, "error creating static pod manifest files for the control plane components")
+		}
+
+		if cfg.Etcd.External == nil {
+			// Checks that the etcd cluster is healthy
+			// NB. this check cannot be implemented before because it requires the admin.conf and all the certificates
+			//     for connecting to etcd already in place
+			kubeConfigFile := filepath.Join(kubeadmconstants.KubernetesDir, kubeadmconstants.AdminKubeConfigFileName)
+
+			client, err := kubeconfigutil.ClientSetFromFile(kubeConfigFile)
+			if err != nil {
+				return errors.Wrap(err, "couldn't create Kubernetes client")
+			}
+
+			if err := etcdphase.CheckLocalEtcdClusterStatus(client, cfg); err != nil {
```

Comment:
Maybe we can use `return etcdphase.CheckLocalEtcdClusterStatus(client, cfg)`

Reply:
I would prefer not to do this here, and only return in case of error in this case, just to keep the flow in case this function gets extended after this block.